src/Symfony/Component/HttpKernel/Controller/ArgumentResolverManager.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\ArgumentResolverInterface;
+
+/**
+ * The ArgumentResolverManager chains over the registered argument resolvers to
+ * resolve all controller arguments.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class ArgumentResolverManager
+{
+    /**
+     * @var array
+     */
+    private $resolvers = array();
+
+    /**
+     * @var null|ArgumentResolverInterface[]
+     */
+    private $sortedResolvers;
+
+    /**
+     * Adds an argument resolver.
+     *
+     * @param ArgumentResolverInterface $resolver
+     */
+    public function add(ArgumentResolverInterface $resolver, $priority = 0)
```

Comment:
do we really need the priority here ? IMO, we could just rely on having ``->add()`` being called in the right order, as we do in many other places.
The sorting by priority would then be done in the compiler pass at compile time (the priority being used to give control over the order when building the container).

IIRC, the only place where we have the priority handled at runtime is for the event dispatcher, and this is because listeners are not all known at compile time.